### DISCUSSION TITLE
Check out the updated topology.pages.dev

### DISCUSSION THREAD

**Discussion by StevenClontz** [pi-base/data#464] [Show and tell]
@jamesdabbs has made some great progress - if you have a chance, try poking around

https://topology.pages.dev

and let us know what you think. I'm starting to use it to review PRs when I can (particularly helpful as you can filter to the asserted properties and hide the deduced properties).

**Comment by prabau** [comment]
Thanks for fixing several of the issues mentioned in #321.  One remains from that previous discussion:

In the explanation page for a particular property for a particular space when it is deduced (example: https://topology.pages.dev/spaces/S000004/properties/P000056), the links from the Value column don't work for me.  I click on them and nothing happens. If I then click refresh for the page, the page goes to the expected new page.  But then the backward link for going to the previous page does not work.

**Reply by jamesdabbs** [reply]
This should be fixed now (updated in https://github.com/pi-base/web/pull/94).

**Comment by prabau** [comment]
Some other observations:

The expansion of links to properties within text, like "{P17}" for example, is not properly displaying any math within dollar signs.  Compare https://topology.pi-base.org/properties/P000025 and https://topology.pages.dev/properties/P000025.

(On the other hand, thank you for properly displaying math within properties aliases now.)

Too much spacing between properties in the general properties list.  (in part, maybe because of unnecessary extra text at the end, and also too much whitespace in the column for propery names?)  Compare: https://topology.pi-base.org/properties and https://topology.pages.dev/properties.

This excessive whitespacing seems to also appear in other lists.  For example, compare https://topology.pages.dev/spaces?q=Zero+Dimensional%2B%7EScattered and https://topology.pi-base.org/spaces?q=Zero+Dimensional%2B%7EScattered.

**Reply by jamesdabbs** [reply]
Added https://github.com/pi-base/web/issues/95 in the `Prerelease` milestone to track the `{P17}` expansion issue. I'm afraid that'll be rather obnoxiously difficult, but is worth doing.

The whitespace changes look to be a few different things:

* For the "list all {space / property / theorem}" pages, the logic around where we cut off the description for previews has changed. The previous version would never include more than one `<p>`aragraph; the new one will.
* For the search page, the whitespace difference looks to come from changes to the SVG icons we're using for checks and x's.

Both of those changes were incidental consequences of some low-level changes, and not anything terribly deliberate. But also, it's not clear to me that they are _worse_ (and #openness pixel-perfect backwards compatibility is a _lot_ of work, and not necessarily a goal). @StevenClontz @prabau – how would y'all feel if we?:

* [x] ~Created an Issue to~ update the list page truncation to always stop at the end of the first paragraph.
* [ ] Leave the icon-spacing whitespace changes as-is for now.

**Reply by jamesdabbs** [reply]
✅ [Here's a PR for some of the truncation content changes](https://github.com/pi-base/web/pull/96).

**Reply by prabau** [reply]
We don't need pixel-perfect backwards compatibility.  But all the extra spacing decreases the information density (i.e. the old version was displaying more information per page) and that's worth restoring in some way (even if the layout changes).

I think stopping at the end of the first paragraph is fine, as all we need a part of the text as a hint of what the property is about.  But note that the display of no more than one paragraph does not fully explain the extra spacing.

In particular, compare https://topology.pi-base.org/properties and https://topology.pages.dev/properties and look for example at property 4 ($T_{2.5}$) which display exactly the same text in the two versions: "_Distinct points are separated by closed neighborhoods. In other words, given any two distinct points..._" but the new version occupies two lines instead of one in the old version.  The reason is that the Description column starts more to the right in the new version.  A hopefully easy fix is to have the Description column start more to the left.

As far as the check/cross icons are concerned, the old icons seemed fine to me.  Not sure why we need new icons of larger size, or we could have new smaller icons.  But if it's too difficult to change, don't bother.  There are more important issues to tackle.

**Reply by jamesdabbs** [reply]
Yeah, this feedback totally makes sense. I'm not sure _why_ the table column spacing is different now, but will spend a little time investigating. I've added a [catchall styling ticket to investigate these](https://github.com/pi-base/web/issues/98).

**Reply by jamesdabbs** [reply]
`{P#}` internal links with embedded math – as in https://topology.pages.dev/properties/P000025 – should render as of https://github.com/pi-base/web/pull/99.

**Reply by prabau** [reply]
Looks good for the `{P#}`

**Comment by prabau** [comment]
The navigation between pages in the browser has problems.  Example: go to https://topology.pages.dev/spaces?q=scattered.  Then click Advanced.  All good so far.  Then try to go back to previous page, and it does not work.  And the previous search is now lost.

**Reply by jamesdabbs** [reply]
Tracking in [this issue](https://github.com/pi-base/web/issues/77)

**Reply by jamesdabbs** [reply]
I think this is fixed now. Mind checking it out and letting me know if you're still hitting issues?

**Reply by prabau** [reply]
That looks good now!

**Comment by prabau** [comment]
Going to a space page like https://topology.pages.dev/spaces/S000078, enter a string in the Filter box, like "compact".  Then click on something in the Value column.  Then go back to the previous page in the browser.  The previous item in the Filter box is lost.  This used to work fine before.

**Reply by StevenClontz** [reply]
On the other hand, current behavior at https://topology.pi-base.org/spaces/S000004 is to log every keystroke in the browser history, making https://topology.pi-base.org/spaces/S000004/properties?filter=compact take eight "back"s to leave the page (if "compact" was typed into the filter box).

Ideally, we should push the filter to history so it appears when going back, but only when navigating away from the page so we don't log every partial search.

**Reply by prabau** [reply]
Yeah, good point.  That's related to the fact that every key stroke in the filter box generated a new url query.  An option would be to do the url query only when the user presses the search button after filling out something in the filter box.  Maybe not as nice for the user though.

Question: do you know if all these url queries go all the way to topology.pi-base.org, or is it a local thing in the browser?

**Reply by StevenClontz** [reply]
All the work is done in the browser - it fetches the entire pi-Base dataset when you load the app. So changing URLs isn't actually communicating with a server - you visit topology.pi-base.org/whatever, and the /whatever route just determines where you start in the app, if that makes sense.

**Reply by jamesdabbs** [reply]
I _think_ this one is going to be the same issue as https://github.com/pi-base/web/issues/77.

**Reply by jamesdabbs** [reply]
Similarly, I think this is fixed. Mind testing it out and letting me know if you're still seeing issues?

**Reply by prabau** [reply]
No, this is not fixed.  Do exactly what's described in the issue at the top of this subconversation and you will see there is no change in behavior yet.  I.e., the previous item in the Filter box is lost when going back.

**Reply by jamesdabbs** [reply]
Sorry, I read this as about `/spaces/all`, not `/spaces/S#`. The related trait tab on individual space and property show pages should now be updated as well.

**Reply by prabau** [reply]
Looks good now.  Thank you!

**Comment by prabau** [comment]
Compare for example these theorem pages (all good):
- https://graph.topology.pages.dev/theorems/T000019
- https://graph.topology.pages.dev/theorems/T000022

with these:
- https://graph.topology.pages.dev/theorems/T000023
- https://graph.topology.pages.dev/theorems/T000164

These latter have weird formatting in the title and also in the text when displaying the converse.

**Reply by jamesdabbs** [reply]
Added to https://github.com/pi-base/web/issues/98; will investigate.

**Reply by jamesdabbs** [reply]
Oh – this may be handled already. https://topology.pages.dev/theorems/T000023 (the primary development branch) looks better than https://graph.topology.pages.dev/theorems/T000023. Is that looking like you'd expect at this point?

**Reply by prabau** [reply]
Yeah, that looks good.  Thanks.

**Comment by prabau** [comment]
One minor user interface thing.  In the properties list for a specific space, like https://topology.pages.dev/spaces/S000007, to the right of the Filter is a box with label that can be toggled between "Showing (robot icon)" and "Hiding (robot icon)".

I think it may be slightly easier for the user to change it to "Hide (robot icon)" and "Show (robot icon)".  In other words, tell what action would happen if the user clicks on the label, instead of says what it's currently doing, which does not make it immediately clear that it's togglable.
What do you think?

**Reply by jamesdabbs** [reply]
Yeah, that makes sense to me. Filed [this issue](https://github.com/pi-base/web/issues/101) – @StevenClontz would you mind confirming that change makes sense (and, if you'd like, make the change)?

**Reply by StevenClontz** [reply]
https://github.com/pi-base/web/pull/103

**Comment by prabau** [comment]
Thanks James for fixing most of the remaining issues.  So it doesn't get lost, I'll repeat the only remaining one I have at this point:

Compare https://topology.pi-base.org/properties and https://topology.pages.dev/properties and look for example at property 4 ($T_{2.5}$) which display exactly the same text in the two versions: "_Distinct points are separated by closed neighborhoods. In other words, given any two distinct points..._" but the new version occupies two lines instead of one in the old version.  The reason is that the Description column starts more to the right in the new version.  A hopefully easy fix is to have the Description column start more to the left.

**Reply by jamesdabbs** [reply]
You're the real MVP. I'm not sure _why_ that's different in this version, so it's another one that's hard to match up exactly. But [this PR](https://github.com/pi-base/web/pull/104) at least tweaks the styling to give the description columns a little more room to breathe.

https://jcd-list-table-widths.topology.pages.dev/properties

**Reply by prabau** [reply]
Thanks and sorry for doing this to you again.  I displayed side by side (1) the old version, (2) the new version and (3) the new version with this latest PR.  

My main complaint was with the list of Properties.  The latest PR changed the spacing in that list to 5%, 20%, 75%.  This gives less space to the Name column than in the version (1) and things start looking too tight when the name is long.  Could we change it to 5%, 25%, 70%? That would bring it very close to version (1) and be quite readable I think.

The Spaces list did not exist in version (1), to nothing to compare there.  But I have to say version (2) seems a lot more readable than version (3) as in version (3) most of the space names are split among multiple lines because the name is usually long.  I would vote for keeping version (2) and not doing any specific size adjustment there.

For the Theorems list I am not sure. (3) is not bad, but maybe (2) was ok too.  I never look at the Description column in the theorems list; if I am interested in a specific theorem, I click on it to get details.

@StevenClontz and others, would you mind giving your opinion on choices (2) vs. (3) for Theorems list in particular?

**Reply by StevenClontz** [reply]
To be honest I don't have strong feelings about the different list styles; perhaps this is an enhancement that could be deferred to after the redesign launch.

I do agree that when I looked at the theorem list, I wondered if the description adds much there, so I made a demo of what that would look like at https://no-thm-description.topology.pages.dev/theorems - I'll merge if @prabau agrees.

**Reply by prabau** [reply]
Hmm, I find that latest version the hardest to read, not because it's missing the description but because the "If" and "Then" columns are really too far apart.  When scanning the list, the eyes have to move left and right and left and right on every line  just to grasp what each line is about.  This is too much eye movement to be comfortable.  But putting the two columns much closer to each other would be fine with me.

If that's too much to do right now, in the interest of moving things along, can we do the following:
- Properties list: change spacing to 5%, 25%, 70%
- Spaces list: keep version (2) = no special spacing specified
- Theorems list: up to James

**Reply by StevenClontz** [reply]
> because the "If" and "Then" columns are really too far apart.

In this case I think we keep Theorems as it is and I'll close that PR.

**Comment by StevenClontz** [comment]
As of today's community meeting, all issues here are resolved or will be resolved by pending PRs.
